aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/transfer/internal/DownloadImpl.java

```diff
@@ -77,7 +74,7 @@ public synchronized ObjectMetadata getObjectMetadata() {
         if (s3Object != null) {
             return s3Object.getObjectMetadata();
         }
-        return objectMetadata;
+        return null;
```

Comment:
No, the s3Object is only populated for serial downloads. For parallel downloads, the s3Object will always be null. That is the reason we store object metadata and return it for parallel downloads.